### PR TITLE
Enable all files_versions features when groupfolders is enabled

### LINKED ISSUE #39137: Make persistent versioning available even with groupfolders being activated
* Provide persistent versioning even with groupfolders being activated, yet just as is for non-groupfolders
* might be covered via groupfolders support [#39138](https://github.com/nextcloud/server/issues/39138)

### PR DESCRIPTION
**Pull request by artonge**
Fix https://github.com/nextcloud/server/issues/39137

-- |Before | After
-- | -- | --
In plain folder | Screenshot from 2023-07-05 15-54-16 | Screenshot from 2023-07-05 15-52-01
In group folder | Screenshot from 2023-07-05 15-54-22 | Screenshot from 2023-07-05 15-52-05

### CODE REVIEW

File: apps/files_versions/lib/Capabilities.php [hunk 1]
```diff
@@ -44,13 +44,11 @@ public function __construct(
 	 * Return this classes capabilities
 	 */
 	public function getCapabilities() {
-		$groupFolderInstalled = $this->appManager->isInstalled('groupfolders');
-
 		return [
 			'files' => [
 				'versioning' => true,
-				'version_labeling' => !$groupFolderInstalled && $this->config->getSystemValueBool('enable_version_labeling', true),
-				'version_deletion' => !$groupFolderInstalled && $this->config->getSystemValueBool('enable_version_deletion', true),
```
Comment (artonge) [inline apps/files_versions/lib/Capabilities.php:53]: @nextcloud/designers, should we drop those settings?

**Review by Pytal — APPROVED** [review]

File: apps/files_versions/lib/Capabilities.php (hunk 1 above)
Comment (artonge) [reply apps/files_versions/lib/Capabilities.php:53]: @szaimen or @nimishavijay ?

Comment (nimishavijay) [reply apps/files_versions/lib/Capabilities.php:53]: Is there any particular reason to drop these settings? Version labelling seems like a pretty important feature, right? :)

**Review by nfebe — APPROVED** [review]

File: apps/files_versions/lib/Capabilities.php (hunk 1 above)
Comment (artonge) [reply apps/files_versions/lib/Capabilities.php:53]: Yes, but should an admin be able to disable labelling and deletion ?

Comment (tiredofit) [reply apps/files_versions/lib/Capabilities.php:53]: I had thought that this PR was meant to _enable_ named versions when in a Group Folder. Is this just a PR to remove the functionality entirely even though it was touted as a launch day feature of NC 26?

**Pull request merged**